test/stability/testDivide/testDivide.go

```diff
@@ -394,93 +398,106 @@ func generatePacket(pkt *packet.Packet, context flow.UserContext) {
-		return
-	}
-	if atomic.AddUint64(&recvPackets, 1) > totalPackets {
-		testDoneEvent.Signal()
+	if commonCheck(pkt) == false {
 		return
 	}
-
-	pkt.ParseData()
-	ipv4 := pkt.GetIPv4()
 	udp := pkt.GetUDPForIPv4()
-	recvIPv4Cksum := packet.SwapBytesUint16(packet.CalculateIPv4Checksum(ipv4))
-	recvUDPCksum := packet.SwapBytesUint16(packet.CalculateIPv4UDPChecksum(ipv4, udp, pkt.Data))
-	if recvIPv4Cksum != ipv4.HdrChecksum || recvUDPCksum != udp.DgramCksum {
-		// Packet is broken
-		atomic.AddUint64(&brokenPackets, 1)
-		return
-	}
 	if udp.DstPort != packet.SwapBytesUint16(dstPort1) {
-		println("Unexpected packet in inputFlow1")
-		println("TEST FAILED")
-		atomic.StoreInt32(&passed, 0)
+		println("Unexpected packet in inputFlow1", udp.DstPort)
+		atomic.AddUint64(&brokenPackets, 1)
 		return
 	}
 	atomic.AddUint64(&recvPacketsGroup1, 1)
 }
 
 func checkInputFlow2(pkt *packet.Packet, context flow.UserContext) {
-	if time.Since(progStart) < T || stabilityCommon.ShouldBeSkipped(pkt) {
-		return
-	}
-	if atomic.AddUint64(&recvPackets, 1) > totalPackets {
-		testDoneEvent.Signal()
+	if commonCheck(pkt) == false {
 		return
 	}
-
-	pkt.ParseData()
-	ipv4 := pkt.GetIPv4()
 	udp := pkt.GetUDPForIPv4()
-	recvIPv4Cksum := packet.SwapBytesUint16(packet.CalculateIPv4Checksum(ipv4))
-	recvUDPCksum := packet.SwapBytesUint16(packet.CalculateIPv4UDPChecksum(ipv4, udp, pkt.Data))
-	if recvIPv4Cksum != ipv4.HdrChecksum || recvUDPCksum != udp.DgramCksum {
-		// Packet is broken
-		atomic.AddUint64(&brokenPackets, 1)
-		return
-	}
 	if gTestType == separate &&
 		udp.DstPort != packet.SwapBytesUint16(dstPort2) &&
 		udp.DstPort != packet.SwapBytesUint16(dstPort3) ||
 		gTestType == split &&
 			udp.DstPort != packet.SwapBytesUint16(dstPort2) ||
 		(gTestType == partition || gTestType == pcopy) &&
 			udp.DstPort != packet.SwapBytesUint16(dstPort1) {
-		println("Unexpected packet in inputFlow2")
-		println("TEST FAILED")
-		atomic.StoreInt32(&passed, 0)
+		println("Unexpected packet in inputFlow2", udp.DstPort)
+		atomic.AddUint64(&brokenPackets, 1)
 		return
 	}
 	atomic.AddUint64(&recvPacketsGroup2, 1)
 }
 
+func commonCheck(pkt *packet.Packet) bool {
+	if time.Since(progStart) < T || stabilityCommon.ShouldBeSkipped(pkt) {
+		return false
+	}
+	if atomic.AddUint64(&recvPackets, 1) > totalPackets {
+		testDoneEvent.Signal()
+		return false
+	}
+
+	if pkt.ParseData() < 0 {
+		println("ParseData returned negative value")
+		atomic.AddUint64(&brokenPackets, 1)
+		return false
+	}
+	ipv4 := pkt.GetIPv4()
+	udp := pkt.GetUDPForIPv4()
+	recvIPv4Cksum := packet.SwapBytesUint16(packet.CalculateIPv4Checksum(ipv4))
+	recvUDPCksum := packet.SwapBytesUint16(packet.CalculateIPv4UDPChecksum(ipv4, udp, pkt.Data))
+	ptr := (*packetData)(pkt.Data)
+	ptr.F2 -= shift
+	if recvIPv4Cksum != ipv4.HdrChecksum || recvUDPCksum != udp.DgramCksum ||
+		ptr.F2 != 2 && ptr.F2 != 7 ||
+		ptr.F2 == 2 && ptr.F1 != 1 ||
+		gTestType != pcopy && ptr.F2 == 7 && checkSlice[ptr.F1] != 0 ||
```

Comment:
not obvious at all, add comment and change magics to constants, they are the same in more than one place, should be changed.